src/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/ConnectionService.java

```diff
@@ -290,6 +291,7 @@ public Response testConnection( DatabaseConnection connection ) throws Connectio
   @Consumes( { APPLICATION_JSON } )
   @Facet( name = "Unsupported" )
   public Response updateConnection( DatabaseConnection connection ) throws ConnectionServiceException {
+    new UtilHtmlSanitizer().sanitizeConnectionParameters( connection );
```

Comment:
@YuryBY, why not instantiating the sanitizer just once, when ConnectionService is being initialized?